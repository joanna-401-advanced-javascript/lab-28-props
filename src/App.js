import React from 'react';
import Message from './components/Message/Message';

class App extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      headline: 'Breaking news!',
    }
  }
  render() {
    return (
      <>
        <Message headline={this.state.headline}/>
      </>
    );
  }
}

export default App;
