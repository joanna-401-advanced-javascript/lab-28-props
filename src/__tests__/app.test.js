import React from 'react';
import renderer from 'react-test-renderer';
import App from '../app.js';

describe('<App />', () => {
  it('nothing from this page is rendered at application start', () => {
    const app = shallow(<App />);
    expect(app.find('div').exists()).toBe(false);
  });

  it('state contains headline', () => {
    const app = mount(<App />);
    expect(app.state('headline')).toEqual('Breaking news!');
  });

  it('state does not include anything from child components', () => {
    const app = mount(<App />);
    expect(app.state('text')).toEqual(undefined);
  });

  it('snapshot is rendered correctly', () => {
    const app = renderer
      .create(<App />)
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
