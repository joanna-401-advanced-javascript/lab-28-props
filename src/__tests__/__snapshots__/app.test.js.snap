// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> snapshot is rendered correctly 1`] = `
<div>
  <p>
    Hello World!
  </p>
  <p>
    Breaking news!
  </p>
  <div>
    <p>
      Author: 
      Margaret Atwood
    </p>
    <p>
      Title: 
      Oryx and Crake
    </p>
  </div>
</div>
`;
